### PR TITLE
MAILBOX-333 Notice should handle size units

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: mailbox/plugin/quota-mailing/src/test/java/org/apache/james/mailbox/quota/mailing/subscribers/QuotaThresholdNoticeTest.java [hunk 1]
```diff
@@ -201,6 +201,30 @@ void generateReportShouldOmitCountPartWhenNone() throws Exception {
...
+                .build())
+            .countQuota(Counts._72_PERCENT)
+            .sizeThreshold(HistoryEvolution.higherThresholdReached(sizeThresholdChange, NotAlreadyReachedDuringGracePeriod))
+            .build()
+            .get()
+            .generateReport(fileSystem))
+            .isEqualTo("You receive this email because you recently exceeded a threshold related to the quotas of your email account.\n" +
+                "\n" +
+                "You currently occupy more than 80 % of the total size allocated to you.\n" +
+                "You currently occupy 801 MB on a total of 1 GB allocated to you.\n" +
```
Comment (mbaechler) [inline mailbox/plugin/quota-mailing/src/test/java/org/apache/james/mailbox/quota/mailing/subscribers/QuotaThresholdNoticeTest.java]: In this particular case, it's 801 MiB (mebibyte not megabyte, see https://en.wikipedia.org/wiki/Byte ), same for GiB.

Comment (mbaechler) [reply mailbox/plugin/quota-mailing/src/test/java/org/apache/james/mailbox/quota/mailing/subscribers/QuotaThresholdNoticeTest.java]: and a substring matching should be enough

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
test this please

File: mailbox/plugin/quota-mailing/src/main/java/org/apache/james/mailbox/quota/mailing/subscribers/SizeFormat.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            if (bytesCount / FileUtils.ONE_GB > 0) {
+                return Optional.of(GiB);
+            }
+            if (bytesCount / FileUtils.ONE_MB > 0) {
+                return  Optional.of(MiB);
+            }
+            if (bytesCount / FileUtils.ONE_KB > 0) {
+                return  Optional.of(KiB);
+            }
+            return Optional.empty();
```
Comment (aduprat) [inline mailbox/plugin/quota-mailing/src/main/java/org/apache/james/mailbox/quota/mailing/subscribers/SizeFormat.java]: Not _Byte_?

File: mailbox/plugin/quota-mailing/src/main/java/org/apache/james/mailbox/quota/mailing/subscribers/SizeFormat.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+        private String asString(BigDecimal bigDecimal) {
+            return DECIMAL_FORMAT.format(bigDecimal.doubleValue());
+        }
+    }
+
+    public static String format(long bytesCount) {
+        Preconditions.checkArgument(bytesCount >= 0, "Formatting of a negative size is forbidden");
+
+        BigDecimal sizeAsDecimal = new BigDecimal(BigInteger.valueOf(bytesCount));
```
Comment (aduprat) [inline mailbox/plugin/quota-mailing/src/main/java/org/apache/james/mailbox/quota/mailing/subscribers/SizeFormat.java]: _BigDecimal_ as a constructor taking a _long_

File: mailbox/plugin/quota-mailing/src/main/java/org/apache/james/mailbox/quota/mailing/subscribers/SizeFormat.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+        return Unit.locateUnit(sizeAsDecimal.longValue())
```
Comment (aduprat) [inline mailbox/plugin/quota-mailing/src/main/java/org/apache/james/mailbox/quota/mailing/subscribers/SizeFormat.java]: Maybe you can use _bytesCount_ as the parameter.

File: mailbox/plugin/quota-mailing/src/main/java/org/apache/james/mailbox/quota/mailing/subscribers/SizeFormat.java (hunk 2 above)
Comment (aduprat) [reply mailbox/plugin/quota-mailing/src/main/java/org/apache/james/mailbox/quota/mailing/subscribers/SizeFormat.java]: Does it make sense to have a unitless long?

File: mailbox/plugin/quota-mailing/src/main/java/org/apache/james/mailbox/quota/mailing/subscribers/QuotaThresholdNotice.java [hunk 5]
```diff
@@ -40,6 +40,7 @@
 import org.apache.james.mailbox.quota.model.HistoryEvolution;
 import org.apache.james.mailbox.quota.model.QuotaThreshold;
 import org.apache.james.mailbox.quota.model.QuotaThresholdChange;
+import org.apache.james.util.SizeFormat;
```
Comment (aduprat) [inline mailbox/plugin/quota-mailing/src/main/java/org/apache/james/mailbox/quota/mailing/subscribers/QuotaThresholdNotice.java]: Is this import needed?

Comment (chibenwa) [reply mailbox/plugin/quota-mailing/src/main/java/org/apache/james/mailbox/quota/mailing/subscribers/QuotaThresholdNotice.java]: Yes

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**